naptime/src/main/scala/org/coursera/naptime/errors.scala

```diff
@@ -46,6 +46,16 @@ case class NaptimeActionException(
       HttpEntity.Strict(ByteString.fromString(bodyString), Some(MimeTypes.JSON)))
   }
 
+  /**
+   * Set detail in NaptimeActionException which can be used for reasoning for downstream service.
+   * TODO: only exposing detail to internal users?
+   * @return NaptimeActionException with details set up as json formated detail.
```

Comment:
I edited the docstring for clarity:
```
* Add exception details which can be used by downstream services for debugging to NaptimeActionException.
* @return A new NaptimeActionException with exception details formatted as JSON.
```

Delete the TODO on internal services. Naptime has no concept of internal and external services.